### ISSUE TITLE
16.9.4, battery level "?" <=> 0 %

### ISSUE THREAD

**Issue by promeneur** [BlissOS/bug_reports#15]
The script to launch the 16.9.4

```
#!/bin/bash

qemu-kvm -enable-kvm \
-m 8G -smp 4 -cpu host \
-display gtk,gl=on,zoom-to-fit=off \
-device virtio-vga-gl \
-device qemu-xhci,p2=5,p3=5,id=xhci \
-device usb-kbd -k fr \
-device usb-mouse \
-device usb-host,vendorid=0x046d,productid=0x0825 \
-device usb-host,vendorid=0x0b05,productid=0x190e \
-device intel-hda \
-device hda-duplex,audiodev=snd0 \
-audiodev pipewire,id=snd0 \
-boot menu=on \
-nic bridge,mac=00:16:3e:4f:90:5a \
~/QEMU_VM/Bliss-v16.9.4-x86_64-OFFICIAL-gapps-20240114.img \

```
the battery level is 0 % with "?" in battery symbol.

**Comment by hmtheboy154** [comment]
this is not a bug, you device doesn't have a battery, of course it will said "?"

Check the Documentation on how to spoof the battery
https://docs.blissos.org/configuration/configuration-through-command-line-parameters/#battery-stats

**Comment by promeneur** [comment]
>> you device doesn't have a battery, of course it will said "?"

off course ? 
No. With android-x86 (android 8.1) there is no problem. The battery symbol indicates that the device is charging and level is 100 % without doing any settings.

Please I get a little knowledge about Bliss, grub, menu entry, kernel command.

There is no more menu.lst in grub folder.
In what command line to add "SET_FAKE_BATTERY_LEVEL" ?

Thanks

**Comment by hmtheboy154** [comment]
> No. With android-x86 (android 8.1) there is no problem.

That is Android 8.1, this is Android 13, things changed



> There is no more menu.lst in grub folder.
> In what command line to add "SET_FAKE_BATTERY_LEVEL" ?

Press E on the entry, edit it and use Ctrl+X to boot

**Comment by promeneur** [comment]
>> Press E on the entry, edit it and use Ctrl+X to boot

It is not permanent.

**Comment by Xtr126** [comment]
> > > Press E on the entry, edit it and use Ctrl+X to boot
> 
> It is not permanent.

https://docs.blissos.org/knowledgebase/troubleshooting/edit-grub-from-bliss-install/

**Comment by promeneur** [comment]
The doc says we must modify android.cfg but look in the contents.
A non-techie person does not understand where to add "SET_FAKE_BATTERY_LEVEL=100". The doc is for a developer.

Thanks

```# $1 Kernel dir
# $2 Title
# $3... Kernel cmdline
function add_boot_entry {
	menuentry "$2" "$@" --class android-x86 {
		savedefault
		set root=$android
		if [ ! -e $2/kernel ]; then
			search --no-floppy --set root -f $2/kernel
		fi
		set kd=$2
		shift 3
		linux $kd/kernel stack_depot_disable=on cgroup_disable=pressure root=/dev/ram0 noexec=off $src $@
		initrd $kd/initrd.img
	}
}

# $1 Title
# $2... Kernel cmdline
function add_entry {
	set title="BlissOS-16.9.4 2024-01-15 $1"
	shift 1
	add_boot_entry "$kdir" "$title" "$@"
}

# $1 Disk number
# $2 partition
function add_winboot_entry {
	menuentry "Windows" {
		insmod part_msdos
		insmod ntfs
		set root="(hd$1,$2)"
		chainloader +1
	}
}

# $1 EFI to chainload
# $2 OS name
# $3 Class
function add_os_if_exists {
	# Is there a better way to find ESP?
	for d in hd0,gpt1 hd0,gpt2 hd1,gpt1 hd1,gpt2 hd0,msdos1 hd0,msdos2 hd1,msdos1 hd1,msdos2; do
		if [ "($d)$1" != "$cmdpath/$bootefi" -a -e ($d)$1 ]; then
			menuentry "$2 at $d ->" "$d" "$1" --class "$3" {
				savedefault
				set root=$2
				chainloader ($root)$3
			}
			break
		fi
	done
}

function savedefault {
	if [ -s $prefix/grubenv -a "$chosen" != "$default" ]; then
		set default="$chosen"
		save_env default
	fi
}

function load_theme {
	loadfont dejavu_14 terminus-14
	set gfxmode=1024x768
	terminal_output gfxterm
	set theme=$prefix/theme/theme.txt
	export theme
}

if [ "$root" == "loop0" ]; then
	set prefix=($root)/boot/grub
fi

if [ -s $prefix/theme/theme.txt ]; then
	load_theme
fi

if [ -s $prefix/grubenv ]; then
	load_env
fi

if [ "$grub_cpu" = "i386" ]; then
	set bootefi=bootia32.efi
	set grub=grubia32
else
	set bootefi=BOOTx64.EFI
	set grub=grubx64
fi

if [ -z "$src" -a -n "$isofile" ]; then
	set src=iso-scan/filename=$isofile
fi

search --no-floppy --set android -f $kdir/kernel
export android bootefi grub kdir live src

# Create main menu
add_entry "$live" quiet
add_entry "$live (Default) w/ FFMPEG" quiet FFMPEG_CODEC=1 FFMPEG_PREFER_C2=1
add_entry "$live PC-Mode (Default)" quiet PC_MODE=1
add_entry "$live PC-Mode (Default) w/ FFMPEG" quiet PC_MODE=1 FFMPEG_CODEC=1 FFMPEG_PREFER_C2=1
#WINDOWS_ENTRY

if [ -s ($android)$kdir/install.img ]; then
	add_entry "Installation" INSTALL=1
fi

submenu "VM Options -> " --class forward {
	add_entry "$live - QEMU/KVM - Virgl - SW-FFMPEG" quiet HWC=drm_minigbm GRALLOC=minigbm_arcvm
	add_entry "$live - Vbox/VMWare - No HW Acceleration" quiet nomodeset HWACCEL=0
	add_entry "$live - Debug QEMU/KVM - Virgl - SW-FFMPEG" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm_arcvm FFMPEG_HWACCEL_DISABLE=1
	add_entry "$live - Debug Vbox/VMWare - No HW Acceleration" DEBUG=2 nomodeset HWACCEL=0
}

submenu "Debugging -> " --class forward {
	add_entry "$live - Debug" DEBUG=2
	add_entry "$live - Debug gralloc.gbm" DEBUG=2 GRALLOC=gbm
	add_entry "$live - Debug drmfb-composer" DEBUG=2 HWC=drmfb GRALLOC=gbm
	add_entry "$live - Debug hwcomposer.drm" DEBUG=2 HWC=drm GRALLOC=gbm
	add_entry "$live - Debug gralloc.minigbm" DEBUG=2 GRALLOC=minigbm
	add_entry "$live - Debug gralloc.minigbm_gbm_mesa" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm_gbm_mesa
	add_entry "$live - Debug hwcomposer.drm_minigbm" DEBUG=2 HWC=drm_minigbm GRALLOC=minigbm
	add_entry "$live - Debug hwcomposer.drm_minigbm_celadon" DEBUG=2 HWC=drm_minigbm_celadon GRALLOC=minigbm
	add_entry "$live - Debug hwcomposer.intel" DEBUG=2 HWC=intel GRALLOC=intel
}

submenu "Advanced options -> " --class forward {
	add_entry "$live - No Setup Wizard" quiet SETUPWIZARD=0
	add_entry "$live - No Hardware Acceleration" quiet nomodeset HWACCEL=0
	if [ -s ($android)$kdir/install.img ]; then
		add_entry "Auto Install to specified harddisk" AUTO_INSTALL=0
		add_entry "Auto Update" AUTO_INSTALL=update
	fi
	add_os_if_exists /EFI/BOOT/$bootefi "UEFI OS" os
	add_os_if_exists /EFI/BOOT/fallback.efi "UEFI Fallback" os
	add_os_if_exists /EFI/BOOT/fallback_x64.efi "UEFI Fallback" os
	menuentry "Reboot" --class reboot { reboot }
	menuentry "Poweroff" --class shutdown { halt }
}

# Add other OSes boot loaders if exist
add_os_if_exists /EFI/fedora/${grub}.efi Fedora fedora
add_os_if_exists /EFI/centos/${grub}.efi CentOS centos
add_os_if_exists /EFI/ubuntu/${grub}.efi Ubuntu ubuntu
add_os_if_exists /EFI/debian/${grub}.efi Debian debian
add_os_if_exists /EFI/gentoo/${grub}.efi Gentoo gentoo
add_os_if_exists /EFI/opensuse/${grub}.efi openSUSE opensuse
add_os_if_exists /EFI/linuxmint/${grub}.efi "Linux Mint" linuxmint
add_os_if_exists /EFI/boto/bootx64.efi OPENTHOS openthos
add_os_if_exists /EFI/Microsoft/Boot/bootmgfw.efi Windows windows
add_os_if_exists /EFI/refind/refind_x64.efi "rEFInd Boot Manager" refind
add_os_if_exists /EFI/refind/refind_ia32.efi "rEFInd Boot Manager" refind
add_os_if_exists /EFI/BlissOS/${grub}.efi BlissOS blissos

for d in $cmdpath $prefix; do
	if [ -f $d/custom.cfg ]; then
		source $d/custom.cfg
	fi
done

```

**Comment by hmtheboy154** [comment]
you came here to ask us where is grub.lst and now we give you the configuration file you are whining that this is not for "non-techie" person ???

**Comment by promeneur** [comment]
>> you came here to ask us where is grub.lst and now we give you the configuration file you are whining that this is not for "non-techie" person ???

Yes I know about grub just where to find menu.lst  and how to find a menu entry and to modify the kernel command.
Also for the kernel command I just know I can add "video=1280x800x32" and that's all.
Many times I don't understand the meaning of the parameters I add. I just follow the advice found in internet in a doc or something else.

For each particular problem I encountered, I learned just the little knowledge to fix the problem.

I can fix only the problems I encountered.

I know nothing about Bliss and the other grub config files and how to modify them.

You are not the first which assumes I am a specialist of grub or something else, but I am not.

In android.cfg, in grub.cfg, I am lost. In the previous version of Bliss there were a menu.lst which disappeared.

Thanks
